promql/parse_test.go

```diff
@@ -882,6 +882,19 @@ var testExpr = []struct {
 				mustLabelMatcher(labels.MatchEqual, string(model.MetricNameLabel), "foo"),
 			},
 		},
+	}, {
+		input: `foo{a="b", foo!="bar", test=~"test", bar!~"baz",}`,
```

Comment:
A simpler example can be used. We should also check that a leading comma is rejected.

Reply:
More tests have been added now, including one with a leading comma.